CoreUtilities/src/au/gov/asd/tac/constellation/utilities/text/SpellCheckingTextArea.java

```diff
@@ -0,0 +1,230 @@
+ *
+ * @author Auriga2
+ */
+public class SpellCheckingTextArea extends InlineCssTextArea {
+
+    private static final Preferences PREFERENCES = NbPreferences.forModule(ApplicationPreferenceKeys.class);
+    private final SpellChecker spellChecker = new SpellChecker(this);
+    private final Insets insets = new Insets(4, 8, 4, 8);
+    public static final double EXTRA_HEIGHT = 3;
+
+    private static final String UNDERLINE_AND_HIGHLIGHT_STYLE = "-rtfx-background-color:derive(yellow,-30%);"
+            + "-rtfx-underline-color: red; "
+            + "-rtfx-underline-dash-array: 2 2;"
+            + "-rtfx-underline-width: 2.0;"
+            + "-fx-fill: black;";
+
+    private static final String CLEAR_STYLE = "-rtfx-background-color: transparent;"
+            + "-rtfx-underline-color: transparent;";
+
+    public SpellCheckingTextArea(final boolean isSpellCheckEnabled) {
+        final boolean enableSpellChecking = PREFERENCES.getBoolean(ApplicationPreferenceKeys.ENABLE_SPELL_CHECKING, ApplicationPreferenceKeys.ENABLE_SPELL_CHECKING_DEFAULT) && isSpellCheckEnabled;
+        spellChecker.turnOffSpellChecking(!enableSpellChecking);
+
+        this.setAutoHeight(false);
+        this.setWrapText(true);
+        this.setPadding(insets);
+        final String css = SpellCheckingTextArea.class.getResource("SpellChecker.css").toExternalForm();//"resources/test.css"
+        this.getStylesheets().add(css);
+
+        this.setOnMouseClicked((final MouseEvent event) -> {
+            if (event.getButton() == MouseButton.PRIMARY && event.isStillSincePress()) {
+                spellChecker.checkSpelling();
+                spellChecker.popUpSuggestionsListAction(event);
+            }
+        });
+
+        this.setOnKeyReleased((final KeyEvent event) -> {
+            if (spellChecker.canCheckSpelling(this.getText())) {
+                spellChecker.checkSpelling();
+            }
+        });
+
+        // Set the right click context menu
+        final ContextMenu contextMenu = addRightClickContextMenu(enableSpellChecking);
+        this.setContextMenu(contextMenu);
+    }
+
+    public SpellCheckingTextArea(final boolean isSpellCheckEnabled, final String text) {
+        this(isSpellCheckEnabled);
+        setText(text);
+    }
+
+    public final void setText(final String text) {
+        this.replaceText(text);
+    }
+
+    /**
+     * underline and highlight the text from start to end.
+     */
+    public void highlightText(final int start, final int end) {
+        this.setStyle(start, end, UNDERLINE_AND_HIGHLIGHT_STYLE);
+    }
+
+    /**
+     * Clear any previous highlighting.
+     */
+    public void clearStyles() {
+        this.setStyle(0, this.getText().length(), CLEAR_STYLE);
+    }
+
+    public boolean isWordUnderCursorHighlighted(final int index) {
+        return (this.getStyleOfChar(index) != null ? this.getStyleOfChar(index).equals(UNDERLINE_AND_HIGHLIGHT_STYLE) : false);
```

Comment:
`return this.getStyleOfChar(index) != null && this.getStyleOfChar(index).equals(UNDERLINE_AND_HIGHLIGHT_STYLE)`